contracts/Otoken.sol

```diff
@@ -1,12 +1,173 @@
-/**
- * SPDX-License-Identifier: UNLICENSED
- */
-pragma solidity 0.6.10;
+/* SPDX-License-Identifier: UNLICENSED */
+pragma solidity =0.6.10;
+
+import {ERC20Initializable} from "./packages/oz/upgradeability/ERC20Initializable.sol";
+import {SafeMath} from "./packages/oz/SafeMath.sol";
+import {BokkyPooBahsDateTimeLibrary} from "./packages/BokkyPooBahsDateTimeLibrary.sol";
+import {AddressBookInterface} from "./interfaces/AddressBookInterface.sol";
 
 /**
- *
+ * @title Otoken
+ * @author Opyn
+ * @notice Otoken is the ERC20 token for an option.
+ * @dev The Otoken inherits ERC20Initializable because we need to use the init instead of constructor.
  */
-// solhint-disable-next-line no-empty-blocks
-contract Otoken {
+contract Otoken is ERC20Initializable {
+    using SafeMath for uint256;
+    address public addressBook;
+    address public underlyingAsset;
+    address public strikeAsset;
+    address public collateralAsset;
+
+    uint256 public strikePrice;
+    uint256 public expiry;
+
+    bool public isPut;
+
+    uint256 private constant STRIKE_PRICE_DIGITS = 10**18;
```

Comment:
use 1e18 instead of 10**x everywhere :) @haythem96 thoughts on what we should use to standardize?

Reply:
yeah I agree, IMO 1e18 looks better.

Reply:
KK